### DISCUSSION TITLE
How do I paginate?

### DISCUSSION THREAD

**Discussion by Mkallday** [thunderclient/thunder-client-support#1161] [Q&A]
Can someone point in the right direction? 
Not only does the "@" in the response header array throw me off, I think the terminology in ThunderClient's documentation is different?

`"@attributes": {
    "next": "https://api.lightspeedapp.com/API/V3/Account/{{accountID}}/Item.json?&after=WyIyMDIzLTA1LTEyVDIyOjQzOjIxKzAwOjAwIiwzODcxNV0%3D",
    "previous": ""
  },`
``

**Comment by rangav** [comment]
Hi @Mkallday 

what are you trying to achieve with the response ?

**Reply by Mkallday** [reply]
paginate to get the next 100.. 

### DOCUMENTATION
 > - The response payload of every GET (query) endpoint includes **two metadata attributes, next and previous**, which contain the  > complete **URLs to fetch the adjacent pages**. 
 > - The next URL will proceed forward in the data set, while the previous URL will go backward to the previously requested page. 
 > - To paginate in a given direction, simply send another request to the provided URL.

 ### SAMPLE REQUEST (FIRST PAGE)
```
 $ curl -H "Authorization: Bearer {Access Token}" \
"https://api.lightspeedapp.com/API/V3/Account/{AccountID}/Item.json"
```
 ### SAMPLE RESPONSE
```
{
    "@attributes": {
        "next": "https://api.lightspeedapp.com/API/V3/Account/{AccountID}/Item.json?sort=itemID&limit=100&*fter=WzEwMF0%3D",
        "previous": ""
    },
    "Item": [
        "..."
    ]
}
```
 ### SAMPLE REQUEST (LAST PAGE)
```
$ curl -H "Authorization: Bearer {Access Token}" \
"https://api.lightspeedapp.com/API/V3/Account/{AccountID}/Item.json?sort=itemID&limit=100&after=WzIwMF0%3D"
```

**Reply by rangav** [reply]
You can save the paging url in the response to env variable and use that variable in the next request.

![image](https://github.com/rangav/thunder-client-support/assets/8637550/33e27cab-2597-4479-9b88-856e207ae4ab)

**Reply by Mkallday** [reply]
You're the best
